pkg/master/pod_cache.go

```diff
@@ -221,6 +227,10 @@ func (p *PodCache) GarbageCollectPodStatus() {
 // calling again, or risk having new info getting clobbered by delayed
 // old info.
 func (p *PodCache) UpdateAllContainers() {
+	// TODO: this is silly, we should pro-actively update the pod status when
+	// the API server makes changes.
+	p.clearNodeStatus()
```

Comment:
Actually, no, now that I look at it, this will fix things.

The problem before was that because the nodeStatus cache was never cleared, we never re-polled node status, so we would never detect the node status change.

Now, when we re-poll the pod status, we will go to the nodeStatus cache, but it will be empty, so we will drive down to the node itself and get the status, and then set that into the cache.